packages/auth/src/errors/utils/assertServiceError.ts

```
@@ -0,0 +1,17 @@
+import { AuthError } from '../AuthError';
+import { ServiceError } from '../types/models';
+
+export function assertServiceError(
+	error: unknown
+): asserts error is ServiceError {
+	if (
+		!error ||
+		!((error as ServiceError).name || !(error as ServiceError).message)
+	) {
```

I just want to confirm that we are asserting that if an error does not have *both* name and message then it will be "unknown"

Is there any risk here that we might swallow information on some error that only has one or the other?

---

addressed comment and added unit tests as well